framework/include/problems/FEProblemBase.h

```diff
@@ -127,6 +127,18 @@ enum MooseLinearConvergenceReason
   MOOSE_DIVERGED_PCSETUP_FAILED = -11
 };
 
+enum MoosePicardConvergenceReason
```

Comment:
You should consider the new strong type `enum class` here.

Reply:
I can use this for all three enums here in FEProblemBase.h.